tests/generate_small_sample.py

```diff
@@ -4,113 +4,160 @@
 
-        h = hashlib.sha256(content.encode()).hexdigest()
+        h = source_hash(content)
         if h in hashes:
             continue
         hashes.add(h)
         yield p
 
 
 def test_file(filename: Path):
-    sys.setrecursionlimit(3000)
-
-    if filename.read_text().count("\n") > 50000:
-        return True, filename
+    code = filename.read_text()
 
-    # clear caches
+    # Clear caches because we are analysing a lot of bytecode here and this can result in hight memory usage.
+    # This is usually not a problem for executing, but this usage scenario is different
     linecache.clearcache()
     for cache_name in ("__source_cache_with_lines", "__executing_cache"):
         if hasattr(Source, cache_name):
             delattr(Source, cache_name)
 
     test = TestFiles()
     try:
-        test.check_filename(filename, check_names=True)
+        with open(os.devnull, "w") as dev_null:
+            with contextlib.redirect_stderr(dev_null):
+                with contextlib.redirect_stdout(dev_null):
+                    test.check_filename(filename, check_names=True)
     except:
-        return False, filename
+        return False
+
+    return True
 
-    return True, filename
+
+def map_file(filename: Path):
+    return test_file(filename), filename
 
 
 def main():
 
-    end_time=time.time()+60*60
+    parser = argparse.ArgumentParser(prog="generate_small_samples")
+    parser.add_argument("source_folder", default="tests/samples", nargs="?")
+
+    args = parser.parse_args()
+
+    folder = Path(args.source_folder)
+
+    if not (folder.exists() and folder.is_dir()):
+        print("source_folder has to be an existing directory")
+        exit(1)
+
+    console = Console()
 
-    with get_context("spawn").Pool(maxtasksperchild=100) as p:
+    end_time = time.time() + 60 * 60
 
-        for result, filename in p.imap_unordered(test_file, big_samples()):
+    console.print()
+    console.print(f"Check files in tests/last_samples and {folder}:")
+    console.print()
 
-            break_file=Path(__file__).parent/"break_generate"
-            if break_file.exists():
-                break_file.unlink()
-                sys.exit(0)
+    with Progress() as progress:
 
-            if time.time()>end_time:
-                print("Timeout")
-                sys.exit(0)
+        task_collect = progress.add_task(description="collect files ...", total=None)
 
+        with get_context("spawn").Pool(maxtasksperchild=100) as p:
+            files = list(
+                progress.track(
+                    big_samples(folder),
+                    task_id=task_collect,
+                    description="collect files...",
+                )
+            )
+            progress.reset(task_collect, description="check files...", total=len(files))
 
-            if not result:
-                try:
+            for result, filename in progress.track(
+                p.imap_unordered(map_file, files), task_id=task_collect
+            ):
```

Comment:
```suggestion
            for result, filename in zip(
                progress.track(
                    p.imap_unordered(test_file, files), task_id=task_collect
                ),
                files,
            ):
```

Reply:
imap_unordered does not preserve the order ... this would mix the results and the filenames.

The reason why I want to ignore the order is that a long running check should not block a failing check for too long (that would not be a huge problem but I see no reason to change this).